src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -0,0 +1,144 @@
+        });
+    }
+
+    private getDeviceInfo(id: string, isEmulator: boolean): Promise<DeviceInfo> {
+        return new Promise<DeviceInfo>(async (resolve, reject) => {
+            try {
+                this.adb.setDeviceId(id);
+                const friendlyName = await this.adb.getModel();
+                resolve({ id, isEmulator, friendlyName });
+            } catch (err) {
+                reject(err);
+            }
+        });
+    }
+
+    public getPackageInfo = (deviceId: string): Promise<PackageInfo> => {
+        return new Promise<PackageInfo>(async (resolve, reject) => {
+            try {
+                this.adb.setDeviceId(deviceId);
+                const info: PackageInfo = await this.adb.getPackageInfo(servicePackageName);
+                if (info?.versionCode !== undefined) {
+                    resolve({ versionCode: info.versionCode, versionName: info.versionName });
+                }
+                reject(new Error('Unable to obtain package version information'));
+            } catch (error) {
+                reject(error);
+            }
+        });
+    };
+
+    public getPermissionInfo = (deviceId: string): Promise<PermissionInfo> => {
+        return new Promise<PermissionInfo>(async (resolve, reject) => {
+            try {
+                const dumpsys = 'dumpsys';
+
+                this.adb.setDeviceId(deviceId);
+                let stdout: string = await this.adb.shell([dumpsys, 'accessibility']);
+                if (!stdout.includes('label=Accessibility Insights')) {
+                    reject(new Error('Accessibility Insights for Android Service is not running'));
+                } else {
+                    stdout = await this.adb.shell([dumpsys, 'media_projection']);
+                    const screenshotGranted: boolean = stdout.includes(servicePackageName);
+                    resolve({ screenshotGranted });
+                }
+            } catch (error) {
+                reject(error);
+            }
+        });
+    };
+
+    public installService = (deviceId: string): Promise<void> => {
+        return new Promise<void>(async (resolve, reject) => {
+            try {
+                const pathToApk = './ServiceForAndroid/AccessibilityInsightsforAndroidService.apk';
+                this.adb.setDeviceId(deviceId);
+                const stdout: string = await this.adb.adbExec(['install', '-d', pathToApk]);
+                if (stdout.includes('Success')) {
+                    resolve();
+                } else {
+                    reject(new Error(stdout));
+                }
+            } catch (error) {
+                reject(error);
+            }
+        });
+    };
+
+    public setTcpForwarding = (deviceId: string): Promise<void> => {
+        return new Promise<void>((resolve, reject) => {
+            try {
+                const port: number = 62442;
+                this.adb.setDeviceId(deviceId);
+                this.adb.forwardPort(port, port);
```

Comment:
`forwardPort` is asynchronous; we should be waiting for it before resolving

Reply:
Yup!